src/gauss_adjoint.jl

```diff
@@ -0,0 +1,431 @@
+        paramjac_config = zero(y), zero(y)
+        pf = let f = unwrappedf
+            if DiffEqBase.isinplace(prob) && prob isa RODEProblem
+                function (out, u, _p, t, W)
+                    f(out, u, _p, t, W)
+                    nothing
+                end
+            elseif DiffEqBase.isinplace(prob)
+                function (out, u, _p, t)
+                    f(out, u, _p, t)
+                    nothing
+                end
+            elseif !DiffEqBase.isinplace(prob) && prob isa RODEProblem
+                function (out, u, _p, t, W)
+                    out .= f(u, _p, t, W)
+                    nothing
+                end
+            else
+                !DiffEqBase.isinplace(prob)
+                function (out, u, _p, t)
+                    out .= f(u, _p, t)
+                    nothing
+                end
+            end
+        end
+        pJ = nothing
+    elseif isautojacvec # Zygote
+        paramjac_config = nothing
+        pf = nothing
+        pJ = nothing
+    else
+        pf = DiffEqBase.ParamJacobianWrapper(unwrappedf, tspan[1], y)
+        pJ = similar(u0, length(u0), numparams)
+        paramjac_config = build_param_jac_config(sensealg, pf, y, p)
+    end
+    GaussIntegrand(sol, p, y, λ, pf, f_cache, pJ, paramjac_config,
+                                sensealg, dgdp_cache, dgdp)
+end
+
+# out = λ df(u, p, t)/dp at u=y, p=p, t=t
+function vec_pjac!(out, λ, y, t, S::GaussIntegrand)
+    @unpack pJ, pf, p, f_cache, dgdp_cache, paramjac_config, sensealg, sol = S
+    f = sol.prob.f
+    isautojacvec = get_jacvec(sensealg)
+    # y is aliased
+
+    if !isautojacvec
+        if DiffEqBase.has_paramjac(f)
+            f.paramjac(pJ, y, p, t) # Calculate the parameter Jacobian into pJ
+        else
+            pf.t = t
+            jacobian!(pJ, pf, p, f_cache, sensealg, paramjac_config)
+        end
+        mul!(out', λ', pJ)
+    elseif sensealg.autojacvec isa ReverseDiffVJP
+        tape = paramjac_config
+        tu, tp, tt = ReverseDiff.input_hook(tape)
+        output = ReverseDiff.output_hook(tape)
+        ReverseDiff.unseed!(tu) # clear any "leftover" derivatives from previous calls
+        ReverseDiff.unseed!(tp)
+        ReverseDiff.unseed!(tt)
+        ReverseDiff.value!(tu, y)
+        ReverseDiff.value!(tp, p)
+        ReverseDiff.value!(tt, [t])
+        ReverseDiff.forward_pass!(tape)
+        ReverseDiff.increment_deriv!(output, λ)
+        ReverseDiff.reverse_pass!(tape)
+        copyto!(vec(out), ReverseDiff.deriv(tp))
+    elseif sensealg.autojacvec isa ZygoteVJP
+        _dy, back = Zygote.pullback(p) do p
+            vec(f(y, p, t))
+        end
+        tmp = back(λ)
+        out[:] .= vec(tmp[1])
+    elseif sensealg.autojacvec isa EnzymeVJP
+        tmp3, tmp4 = paramjac_config
+        tmp4 .= λ
+        out .= 0
+        Enzyme.autodiff(Enzyme.Reverse, pf, Enzyme.Duplicated(tmp3, tmp4),
+                        y, Enzyme.Duplicated(p, out), t)
+    end
```

Comment:
we should probably `else error "not supported"`

Reply:
added